spring-integration-core/src/main/java/org/springframework/integration/support/management/MetricsContext.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.integration.support.management;
+
+/**
+ * Interface representing an opaque object containing state between initiating an
+ * event and concluding it.
+ *
+ * @author Gary Russell
+ * @since 4.2
+ *
+ */
+public interface MetricsContext {
```

Comment:
I don't have a real use case in mind, I just wanted to provide the flexibility in case user implementations need to retain more state than a simple long.